src/HTTP/Factory.php

```diff
@@ -0,0 +1,75 @@
+<?php
+/* Copyright (c) 2016 Fabian Schmid <fs@studer-raimann.ch> Extended GPL, see docs/LICENSE */
+namespace ILIAS\HTTP;
+
+use ILIAS\HTTP\Cookies\CookieJar;
+use ILIAS\HTTP\Response\rendering\RenderingException;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+
+/**
+ * Interface Factory
+ *
+ * @author  Fabian Schmid <fs@studer-raimann.ch>
+ * @author  Nicolas Schäfli <ns@studer-raimann.ch>
+ * @since   5.2
+ * @version 1.0.0
+ */
+interface Factory {
+
+	/**
+	 * Returns the current psr-7 server request.
+	 *
+	 * @return ServerRequestInterface
+	 */
+	public function request();
+
+
+	/**
+	 * Returns the current psr-7 response.
+	 *
+	 * @return ResponseInterface
+	 */
+	public function response();
+
+
+	/**
+	 * Returns a cookie jar which has all cookies known by the ILIAS response.
+	 * Make sure to call the saveResponse method when the cookies are rendered into the response
+	 * object.
+	 *
+	 * @return CookieJar
+	 */
+	public function cookieJar();
+
+
+	/**
+	 * Saves the given request for further use.
+	 * The request should only be saved if absolutely necessary.
+	 * There is a possibility that the request can't be saved back in the near future.
+	 *
+	 * @param ServerRequestInterface $request The server request which should be saved.
+	 *
+	 * @return void
+	 */
+	public function saveRequest(ServerRequestInterface $request);
```

Comment:
Hi @d3r1w,

this is the binding to be able to support the current handling of HTTP in ILIAS we once talked about, right?

From the perspective of the proposed FIG-Interfaces for HTTP services, request handling (i.e. our app) should have the form of a function from Request to Response. This makes it possible to e.g. use some 3rd party middleware somewhere in the chain that performs the request handling. The kind of object you are proposing with this `Factory` somehow circumvents this style of request handling, as it introduces the possibility to store some request at some time or get another request somewhere else. This destroys some of the predictability that is implied by the Request -> Response model.
I see, however, how we require such a facility to be able to use this HTTP-service at the moment, i.e. make use of it in a world where the request is stored in some super globals and responses are made via the global template or echo. Maybe we should name this facility `GlobalHTTPState` to have a reminder for this.

Thx for the good work, though.

Regards!